ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt

```diff
@@ -60,14 +64,30 @@ internal constructor(
   val bottomSheetState: @Hot(replays = true) MutableLiveData<BottomSheetState> = MutableLiveData()
   val showLocationOfInterestSelectorRequests: @Hot Subject<ImmutableList<LocationOfInterest>> =
     PublishSubject.create()
+  /**
+   * Live cache of locations of interest. Updated every time the underlying local storage data
+   * changes.
+   */
+  private var locationOfInterestCache: ImmutableSet<LocationOfInterest> = ImmutableSet.of()
 
   fun openNavDrawer() {
     openDrawerRequests.onNext(Nil.NIL)
   }
 
-  fun onMarkerClick(mapLocationOfInterest: MapLocationOfInterest) {
-    showBottomSheet(mapLocationOfInterest.locationOfInterest)
-  }
+  /** Intended for use as a callback for handling user clicks on rendered map features. */
+  fun onMarkerClick(feature: Feature) =
```

Comment:
nit: Let's name it in a more generic way since this would handle not just markers

Reply:
Also, do we actually need this method? In the mocks, only the bottom LOI cards were clickable. @gino-m